src/wp-includes/abilities-api/class-wp-abilities-registry.php

```diff
@@ -0,0 +1,313 @@
+	 *   category?: string,
+	 *   execute_callback?: callable( mixed $input= ): (mixed|\WP_Error),
+	 *   permission_callback?: callable( mixed $input= ): (bool|\WP_Error),
+	 *   input_schema?: array<string,mixed>,
+	 *   output_schema?: array<string,mixed>,
+	 *   meta?: array{
+	 *     annotations?: array<string,(bool|string)>,
+	 *     show_in_rest?: bool,
+	 *     ...<string, mixed>
+	 *   },
+	 *   ability_class?: class-string<\WP_Ability>,
+	 *   ...<string, mixed>
+	 * } $args
+	 */
+	public function register( string $name, array $args ): ?WP_Ability {
+		if ( ! preg_match( '/^[a-z0-9-]+\/[a-z0-9-]+$/', $name ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				__(
+					'Ability name must be a string containing a namespace prefix, i.e. "my-plugin/my-ability". It can only contain lowercase alphanumeric characters, dashes and the forward slash.'
+				),
+				'6.9.0'
+			);
+			return null;
+		}
+
+		if ( $this->is_registered( $name ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Ability name. */
+				sprintf( __( 'Ability "%s" is already registered.' ), esc_html( $name ) ),
+				'6.9.0'
+			);
+			return null;
+		}
+
+		/**
+		 * Filters the ability arguments before they are validated and used to instantiate the ability.
+		 *
+		 * @since 6.9.0
+		 *
+		 * @param array<string,mixed> $args {
+		 *     An associative array of arguments for the ability.
+		 *
+		 *     @type string               $label                 The human-readable label for the ability.
+		 *     @type string               $description           A detailed description of what the ability does.
+		 *     @type string               $category              The category slug this ability belongs to.
+		 *     @type callable             $execute_callback      A callback function to execute when the ability is invoked.
+		 *                                                       Receives optional mixed input and returns mixed result or WP_Error.
+		 *     @type callable             $permission_callback   A callback function to check permissions before execution.
+		 *                                                       Receives optional mixed input and returns bool or WP_Error.
+		 *     @type array<string,mixed>  $input_schema          Optional. JSON Schema definition for the ability's input.
+		 *     @type array<string,mixed>  $output_schema         Optional. JSON Schema definition for the ability's output.
+		 *     @type array<string,mixed>  $meta                  {
+		 *         Optional. Additional metadata for the ability.
+		 *
+		 *         @type array<string,bool|string> $annotations  Optional. Annotation metadata for the ability.
+		 *         @type bool                      $show_in_rest Optional. Whether to expose this ability in the REST API. Default false.
+		 *     }
+		 * }
+		 * @param string              $name The name of the ability, with its namespace.
+		 */
+		$args = apply_filters( 'register_ability_args', $args, $name );
```

Comment:
See above: This here is an example where omitting the `wp_` prefix makes it particularly awkward:
* The function most devs will call is `wp_register_ability`.
* But then the relevant hook is not called `wp_register_ability_args`.

So this is another point for consistently including the `wp_` prefix.